content/english/blog/road_to_ubiquity.md

```diff
@@ -0,0 +1,75 @@
+
+This engine could be optimized and focused on any number of targets or categories like small devices, the cloud, a browser, or other bespoke environments with highly tuned characteristics. In keeping with WebAssembly's portability promise, the choice of engine should never require a refactor, redesign, or even a recompilation.
+
+## Specification-Level Choice
+It seems likely that ultimately WASI will not be a single all-encompassing walled box of extras that developers get to use with their modules, but rather a grab-bag of opt-in choices to multiple standards like networking, cryptography, file system and OS access, video cards/GPUs, etc. It's also pretty likely that these opt-in specifications will be defined as components using the component model.
+
+Developers wishing to take advantage of specific WASI features are going to have to make that choice at design time and this choice gets baked into their WebAssembly module. If a developer chooses to write to stdout via WebAssembly, this may require one component while publishing a message over a message broker topic would be yet another component.
+
+WASI is also likely going to be used as a gap bridging device. People who want the portability of WebAssembly, have little interest in the component model, and still want their big legacy libraries to compile and work[^1] properly will leverage WASI-aware compilers/linkers (like the way Rust/LLVM is today) to enjoy the best of both worlds. The ecosystem will likely have multiple "smoke and mirror" tools that highly leverage WASI in clever ways to further provide "magic" and "hand-waving" to hide complexity from developers and operations. Unfortunately, many of these fancy shims will probably still require a browser.
+
+At the **engine** and **specification** level, there should be no issues around vendor lock-in and forcing developers to choose between clouds, platforms, environments, etc.
+
+## Application-Level Choice
+Today, the choice of application or application platform to support WebAssembly modules is extremely limited. In addition to the lack of choice variety, each choice requires a _total embrace_ and **_lock_** into a particular vendor's environment, SDK, tool chain, and so on.
+
+In the current ecosystem, if you want edge functions written in WebAssembly, you'll have to use _Fastly_ or _CloudFlare_'s SDK. If you want workflows to chain WebAssembly functions, you'll need to use _Atmo_'s SDK. If you want self-contained, freestanding microservices written in WebAssembly, you could use the _Spin_ SDK. If you want portable distributed actors loosely coupled to capability contracts, you may want to use the _wasmCloud_ SDK.
+
+## Embracing the Road to Ubiquity
+As the WebAssembly community and ecosystem moves forward, we (wasmCloud and Cosmonic) need to continue to move at the vanguard of this momentum, so that we're adapting our foundations to incorporate new power, features, and flexibility as it becomes available and more mature.
```

Comment:
I keep going back and forth on if `wasmcloud and cosmonic` is weird on the wasmcloud blog 🤔 I like how it aligns that wasmCloud devs are employed by a venture funded company and the company is behind the standards, too, though.

Reply:
there's quite a bit of precedent of people posting about the commercial sponsor for the open source effort. I look at how NATS and Synadia do it as well as how OPA and Styra do it.